booster/library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -395,7 +395,9 @@ llvmSimplify term = do
   where
     evalLlvm definition api cb t@(Term attributes _)
         | attributes.isEvaluated = pure t
-        | isConcrete t && attributes.canBeEvaluated = withContext CtxLlvm . withTermContext t $ do
+        | isConcrete t
+        , attributes.canBeEvaluated
+        , isFunctionApp t = withContext CtxLlvm . withTermContext t $ do
```

Comment:
This is the important change for performance, the rest is about logging the LLVM time.